### DISCUSSION TITLE
Issue with Variable Message Size in Source Script

### DISCUSSION THREAD

**Discussion by bjmarty122** [USNavalResearchLaboratory/mgen#43] [General]
I attempted to add the uniform random message size option to my script (from the example 0.0 ON 1 UDP DST 127.0.0.1/5000 PERIODIC [1 824:1224]).

Yet, the packets sent are only of the sizeMin (i.e., 824 bytes) and not of a uniform sample from the range minSize to maxSize.

Has anyone else encountered this issue and/or know of a solution?

**Comment by weston-nrl** [comment]
Are you sure you are using the latest MGEN and you don't maybe have two versions installed?  Look at `which mgen` to be sure?  What is your OS?

This works fine for me on Ubuntu:
$ mgen event "on 1 udp dst 127.0.0.1/5000 periodic [1 824:1224]"

$ mgen port 5000
GPSSubscribe(): fopen() error: No such file or directory
GPSSubscribe(): fopen() error: No such file or directory
mgen: version 5.1.1
mgen: starting now ...
15:38:58.640537 START Mgen Version 5.1.1
15:38:58.640671 LISTEN proto>UDP port>5000
15:39:27.057108 RECV proto>UDP flow>1 seq>0 src>127.0.0.1/42492 dst>127.0.0.1/5000 sent>15:39:27.057009 size>1211 gps>INVALID,999.000000,999.000000,4294966297
15:39:28.057193 RECV proto>UDP flow>1 seq>1 src>127.0.0.1/42492 dst>127.0.0.1/5000 sent>15:39:28.057052 size>1076 gps>INVALID,999.000000,999.000000,4294966297
15:39:29.057161 RECV proto>UDP flow>1 seq>2 src>127.0.0.1/42492 dst>127.0.0.1/5000 sent>15:39:29.057037 size>1074 gps>INVALID,999.000000,999.000000,4294966297
15:39:30.057534 RECV proto>UDP flow>1 seq>3 src>127.0.0.1/42492 dst>127.0.0.1/5000 sent>15:39:30.057412 size>1008 gps>INVALID,999.000000,999.000000,4294966297
15:39:31.057468 RECV proto>UDP flow>1 seq>4 src>127.0.0.1/42492 dst>127.0.0.1/5000 sent>15:39:31.057396 size>1145 gps>INVALID,999.000000,999.000000,4294966297
15:39:32.057465 RECV proto>UDP flow>1 seq>5 src>127.0.0.1/42492 dst>127.0.0.1/5000 sent>15:39:32.057393 size>890 gps>INVALID,999.000000,999.000000,4294966297
15:39:33.057528 RECV proto>UDP flow>1 seq>6 src>127.0.0.1/42492 dst>127.0.0.1/5000 sent>15:39:33.057414 size>1213 gps>INVALID,999.000000,999.000000,4294966297
^C15:39:35.530479 STOP

**Reply by bjmarty122** [reply]
I'm using mgen version 5.02b on the latest Raspberry Pi OS (Linux 5.15.61)

**Reply by weston-nrl** [reply]
Well, that'd be the problem.  5.02b is 7 years old, and this is a more recent addition.

**Reply by bjmarty122** [reply]
perfect, thank you! 

Odd that the old version is what installed, considering I only installed it a few weeks ago, but I'll go ahead and install the latest, looks to be 5.1.1
